terminal.h

```diff
@@ -513,18 +512,33 @@ void codepoint_to_utf8(std::string &s, char32_t c) {
         case 1: u1 =  (c | mask[nbytes-1]);
     }
     switch (nbytes) {
-        case 1: s.push_back(u1); break;
-        case 2: s.push_back(u1); s.push_back(u2); break;
-        case 3: s.push_back(u1); s.push_back(u2); s.push_back(u3); break;
-        case 4: s.push_back(u1); s.push_back(u2); s.push_back(u3); s.push_back(u4); break;
+      case 1:
+        s.push_back(static_cast<char>(u1));
+        break;
+      case 2:
+        s.push_back(static_cast<char>(u1));
+        s.push_back(static_cast<char>(u2));
+        break;
+      case 3:
+        s.push_back(static_cast<char>(u1));
+        s.push_back(static_cast<char>(u2));
+        s.push_back(static_cast<char>(u3));
+        break;
+      case 4:
+        s.push_back(static_cast<char>(u1));
+        s.push_back(static_cast<char>(u2));
+        s.push_back(static_cast<char>(u3));
+        s.push_back(static_cast<char>(u4));
+        break;
```

Comment:
And in here we just need to take the `u1`, `u2`, ... and  concatenate  them at the end of the string `s`.

I don't like this code, but I couldn't figure out how to do it better. I just got an idea, something like:
```c++
idx = s.size();
s.append(std::string(nbytes, "X"))
s[idx] = u1; idx++;
if (nbytes >= 2) { s[idx] = u2; idx++; }
if (nbytes >= 3) { s[idx] = u3; idx++; }
if (nbytes >= 4) { s[idx] = u4; }
```
It still feels ugly, but better than the current version. I'll send a separate PR with this.

Reply:
changing the type of u1,u2,u3,u4 to `char` allows me to remove all of these casts.